### PR TITLE
Checking if the string follows the given pattern

### PR DESCRIPTION
**Pull request by darkrishabh**

### CODE REVIEW

File: python/string/stringFollowPattern.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+"""
+Given a string and a pattern, does the string follow the given pattern
+eg - String - "red black red"
+    Pattern - "aba"
+
+    result = True
+"""
+
+
+def wordPattern(str, pattern):
```
Comment (orsenthil) [inline python/string/stringFollowPattern.py]: Don't use a name`str` as it shadows the builtin with the same name.

File: .gitignore [hunk 2]
```diff
@@ -2,6 +2,7 @@ interview.iml
 interview.ipr
 interview.iws
 target/
+python/.idea/
```
Comment (orsenthil) [inline .gitignore]: You could remove this change from this commit.

File: python/string/stringFollowPattern.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    if len(pattern) == 0 or len(str) == 0: return False
+
+    word_list = str.split()
+    pattern_list = list(pattern)
+    if len(word_list) != len(pattern_list): return False
+
+    w2p = {}
+    p2w = {}
```
Comment (orsenthil) [inline python/string/stringFollowPattern.py]: Only one dictionary, `pattern_to_word` should be sufficient. the other is not required here.

**Comment by mission-peace** [comment]
No response for quite some time. Closing.

**Pull request closed without merging**